File: src/Na_code/BigNum_1.java
package Na_code;

import java.util.*;

public class BigNum_1 {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        int repeat = sc.nextInt();
        int max = sc.nextInt();
        ArrayList<Integer> list = new ArrayList<>();
        for(int i =0; i<size; i++)
            list.add(sc.nextInt());
        Collections.sort(list, Comparator.reverseOrder());

        int result =0;
        result+=repeat/(max+1) * (list.get(0) *max + list.get(1));
        result+=repeat%(max+1) * list.get(0);
        System.out.println(result);


    }
}

File: src/Codeup/BigNum_1.py
n, m, k = map(int, input().split())
data = list(map(int, input().split()))

data.sort();
result = 0
result += (int)(m/(k+1) * (data[n-1] * k + data[n-2]))
result += (int)(m % (k+1) * data[n-1])
print(result)

'''
5 8 3
2 4 5 4 6
'''